### DISCUSSION TITLE
Showing data from relationships without redundant schemas

### DISCUSSION THREAD

**Discussion by donnyxray** [filamentphp/filament#6882] [Help] [answered]
I have defined resources for a **User** model and a **Car** model. Each **User** can _HasMany_ **Car** models. When I view a UserResource I want to immediately see a list of Cars belonging to him.

I currently do this using a Repeater with a relationship in the UserResource. It works well, but I don't like two things:

1. It causes me to define the schema for the Cars twice. Once in the CarResource itself, once in the Repeater schema in the UserResource.
2. It displays as a form rather than table. (If you've ever used Nova you'll know what I'm looking for).

Is there a different way to do this? Can I tell the Repeater to simply look in the CarResource for the schema to use? Or somehow extract the schema?

**Comment by pboivin** [comment]
Not sure if I understood correctly but this sounds a bit like a relation manager... Have you explored that?
https://filamentphp.com/docs/2.x/admin/resources/relation-managers

**Comment by donnyxray** [comment]
Thanks! That does indeed help to view it as a table rather than a form. I still need to define the table schema twice though.

For now I work around that by putting `return CarResource::table($table)` in the table function of the CarRelationshipManager.

If anyone has a better way of avoiding redundant definitions, I would love to know.

**Reply by pboivin** [reply, marked as answer]
I think reusing the static `table()` method is a good solution, I do it often.
